ner_v2/detectors/numeral/number/number_detection.py

```diff
@@ -127,7 +130,13 @@ def detect_entity(self, text, **kwargs):
         number_data = self.language_number_detector.detect_number(self.processed_text)
         validated_number, validated_number_text = [], []
         for number_value_dict, original_text in zip(number_data[0], number_data[1]):
-            if self.min_digit <= len(number_value_dict[NUMBER_DETECTION_RETURN_DICT_VALUE]) <= self.max_digit:
+            number_value = number_value_dict[NUMBER_DETECTION_RETURN_DICT_VALUE]
+            number_unit = number_value_dict[NUMBER_DETECTION_RETURN_DICT_UNIT]
+            if self.min_digit <= len(number_value) <= self.max_digit:
+                if self.unit_type and (not self.language_number_detector.units_map.get(number_unit) or
+                                       (self.language_number_detector.units_map.get(number_unit) and
+                                        self.language_number_detector.units_map[number_unit].type != self.unit_type)):
```

Comment:
Why not pass on the unit type to the lower level detectors and exclude irrelevant units from the regex itself? Would make the detection faster.

Reply:
Why is this needed? not self.language_number_detector.units_map.get(number_unit)

Reply:
If case no unit is present in number detected but unit_type is passed in request.